app/components/Home.tsx

```diff
@@ -0,0 +1,368 @@
+import * as React from 'react';
+const { EditorKit, EditorKitDelegate } = require('sn-editor-kit');
+const MarkdownIt = require('markdown-it');
+
+type Mode = {
+  type: ModeType;
+  label: string;
+  css: string;
+};
+
+enum ModeType {
+  Edit = 0,
+  Split = 1,
+  Preview = 2,
+}
+
+enum MouseEvent {
+  Down = 'mousedown',
+  Move = 'mousemove',
+  Up = 'mouseup',
+}
+
+enum HtmlElementId {
+  ColumnResizer = 'column-resizer',
+  Editor = 'editor',
+  EditorContainer = 'editor-container',
+  Header = 'header',
+  Preview = 'preview',
+  SimpleMarkdown = 'simple-markdown',
+}
+
+enum ComponentDataKey {
+  Mode = 'mode',
+}
+
+const modes = [
+  { type: ModeType.Edit, label: 'Edit', css: 'edit' } as Mode,
+  { type: ModeType.Split, label: 'Split', css: 'split' } as Mode,
+  { type: ModeType.Preview, label: 'Preview', css: 'preview' } as Mode,
+];
+
+/** MarkdownIt Options:
+ * html option allows inline HTML
+ * linkify option automatically renders raw links as anchors
+ * breaks option creates new lines without trailing spaces
+ */
+
+const MarkdownItOptions = {
+  html: true,
+  linkify: true,
+  breaks: true,
+};
+
+const MarkdownParser = MarkdownIt(MarkdownItOptions)
+  .use(require('markdown-it-footnote'))
+  .use(require('markdown-it-task-lists'))
+  .use(require('markdown-it-highlightjs'));
+
+type HomeState = {
+  text?: string;
+  mode: Mode;
+  platform?: string;
+};
+
+const debugMode = false;
+
+const keyMap = new Map();
+
+export class Home extends React.Component<{}, HomeState> {
+  editorKit: any;
+  note: any;
+
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      mode: modes[0],
+      text: '',
+    };
+  }
+
+  componentDidMount = () => {
+    this.configureEditorKit();
+    this.configureMarkdown();
+    this.configureResizer();
+  };
+
+  truncateString = (string: string, limit = 80) => {
+    if (!string) {
+      return null;
+    }
+    if (string.length <= limit) {
+      return string;
+    } else {
+      return string.substring(0, limit) + '...';
+    }
+  };
+
+  configureEditorKit() {
+    const delegate = new EditorKitDelegate({
+      setEditorRawText: (text: string) => {
+        this.setState(
+          {
+            text,
+          },
+          () => {
+            this.updatePreviewText();
+            this.loadSavedMode();
+          }
+        );
+      },
+      clearUndoHistory: () => {},
+      getElementsBySelector: () => [] as any,
+      generateCustomPreview: (text: string) => {
+        this.updatePreviewText();
+        const preview = document.getElementById(HtmlElementId.Preview);
+        const previewText = this.truncateString(
+          preview.textContent || preview.innerText
+        );
+        return {
+          html: `<div>${previewText}</div>`,
```

Comment:
We should be returning either `html` or `plain`, but not both. Some editors use HTML, like the Task Editor, to generate rich previews, but this editor doesn't.

Reply:
Oh ok. Why doesn't this editor use HTML? I referenced the Token Vault editor to write this code. It uses both `html` and `plain`:  https://github.com/sn-extensions/token-vault/blob/master/app/components/Home.js#L57

Reply:
Ah ok, you're right, it's possible to use both. This is because on web we support rich previews, but on mobile we need only plain reviews. But for Markdown documents, if you use `# Hello` in the first line, we don't want a huge header in your note preview. So that's why we want plain only.

Reply:
I don't think the huge header in the note preview is an issue here because we are only taking the `textContent` or `innerText` of the Preview HTML, so we're escaping all the HTML tags. See this reference: https://www.w3schools.com/jsref/tryit.asp?filename=tryjsref_node_textcontent_innerhtml_innertext